File: src/intellibitz/sted/actions/LoadFontMapAction.java
package intellibitz.sted.actions;

import intellibitz.sted.event.FontMapChangeEvent;
import intellibitz.sted.event.FontMapChangeListener;
import intellibitz.sted.launch.STEDGUI;
import intellibitz.sted.ui.STEDWindow;

import javax.swing.event.InternalFrameEvent;
import javax.swing.event.InternalFrameListener;
import javax.swing.event.TableModelEvent;
import java.awt.event.ActionEvent;

public class LoadFontMapAction
        extends TableModelListenerAction
        implements FontMapChangeListener,
        InternalFrameListener {
    public LoadFontMapAction() {
        super();
    }

    /**
     * This fine grain notification tells listeners the exact range of cells,
     * rows, or columns that changed.
     */
    public void tableChanged(TableModelEvent e) {
        setEnabled(
                getSTEDWindow().getDesktop()
                        .getFontMap().isReloadable());
    }

    /**
     * Invoked when the target of the listener has changed its state.
     *
     * @param e a ChangeEvent object
     */
    public void stateChanged(FontMapChangeEvent e) {
        setEnabled(e.getFontMap().isReloadable());
    }

    public void actionPerformed(ActionEvent e) {
        final STEDWindow stedWindow = getSTEDWindow();
        STEDGUI.busy();
        stedWindow.getDesktop().reloadFontMap();
        fireStatusPosted("FontMap Re-Loaded");
        STEDGUI.relax();
    }

    public void internalFrameActivated(InternalFrameEvent e) {

    }

    public void internalFrameClosed(InternalFrameEvent e) {

    }

    public void internalFrameClosing(InternalFrameEvent e) {
        setEnabled(false);
    }

    public void internalFrameDeactivated(InternalFrameEvent e) {

    }

    public void internalFrameDeiconified(InternalFrameEvent e) {

    }

    public void internalFrameIconified(InternalFrameEvent e) {

    }

    public void internalFrameOpened(InternalFrameEvent e) {

    }

}

File: src/main/intellibitz/sted/actions/LoadFontMapAction.kt
package sted.actions

import sted.STEDGUI
import sted.event.FontMapChangeEvent
import sted.event.FontMapChangeListener
import java.awt.event.ActionEvent
import javax.swing.event.InternalFrameEvent
import javax.swing.event.InternalFrameListener
import javax.swing.event.TableModelEvent

class LoadFontMapAction : TableModelListenerAction(), FontMapChangeListener, InternalFrameListener {
    /**
     * This fine grain notification tells listeners the exact range of cells,
     * rows, or columns that changed.
     */
    override fun tableChanged(e: TableModelEvent) {
        isEnabled = stedWindow.desktop.fontMap.isReloadable
    }

    /**
     * Invoked when the target of the listener has changed its state.
     *
     * @param fontMapChangeEvent a ChangeEvent object
     */
    override fun stateChanged(fontMapChangeEvent: FontMapChangeEvent) {
        isEnabled = fontMapChangeEvent.fontMap.isReloadable
    }

    override fun actionPerformed(e: ActionEvent) {
        STEDGUI.busy()
        stedWindow.desktop.reloadFontMap()
        fireStatusPosted("FontMap Re-Loaded")
        STEDGUI.relax()
    }

    override fun internalFrameActivated(e: InternalFrameEvent) {}
    override fun internalFrameClosed(e: InternalFrameEvent) {}
    override fun internalFrameClosing(e: InternalFrameEvent) {
        isEnabled = false
    }

    override fun internalFrameDeactivated(e: InternalFrameEvent) {}
    override fun internalFrameDeiconified(e: InternalFrameEvent) {}
    override fun internalFrameIconified(e: InternalFrameEvent) {}
    override fun internalFrameOpened(e: InternalFrameEvent) {}
}